client/question/services/code_evaluators/PythonPrereqCheckServiceSpec.js

```diff
@@ -493,7 +493,33 @@ describe('PythonPrereqCheckService', function() {
       ].join('\n');
       var prereqFailureType =
         PythonPrereqCheckService.detectAndGetWrongLanguageType(code);
-      expect(prereqFailureType).toEqual(null);
+      expect(prereqFailureType).toEqual(
+        jasmine.objectContaining({
+          _errorName: 'notOp',
+          _errorLineNumber: 2,
+          _errorType: 'wrongLang'
+        }));
+    });
+  });
+
+  describe('obscureStringCharacters', function() {
```

Comment:
This method has complicated logic and needs way more tests. Strings within strings, multiple strings, strings of different types, single-and-double-quotes interlacing, etc.

You don't need to use actual "looks like code" strings, btw. Any strings will do.

Reply:
Added new test conditions to flesh out functionality.